File: M1/Assignment1.R
set.seed(123)

# Dynamically create a list of 400 workers 
workers <- list()
for (i in 1:400) {
  worker <- list(
    id = i,
    name = paste("Worker", i, sep = "_"),
    salary = sample(1000:50000, 1),
    sex = sample(c("Male", "Female"), 1)
  )
  workers[[i]] <- worker
}

# Generate payment slips for each worker
for (worker in workers) {
  tryCatch({
    salary <- worker$salary
    sex <- worker$sex
    
    # Apply the conditional statements
    if (salary > 7500 & salary < 30000 & sex == "Female") {
      worker$level <- 'A5-F'
    } else if (salary > 10000 & salary < 20000) {
      worker$level <- 'A1'
    } else {
      worker$level <- 'Other' # Default level if no conditions are met
    }
    
    # Print payment slips
    cat("Payment Slip: ID=", worker$id, ", Name=", worker$name, 
        ", Salary=", worker$salary, ", Sex=", worker$sex,
        ", Level=", worker$level, "\n", sep = "")
  }, 
  
  # Exception handling for potential errors
  error = function(e) {
    cat("Error processing worker", worker$id, ":", conditionMessage(e), "\n")
  })
}

File: M1/Assignment1.py
from random import randint, choice

# Dynamically create a list of 400 workers 
workers = []
for i in range(1, 401):
    worker = {
        'id': i,
        'name': f'Worker_{i:000}',
        'salary': randint(1000, 50000),
        'sex': choice(['Male', 'Female'])
    }
    workers.append(worker)

# Generate payment slips for each worker
for worker in workers:
    try:
        salary = worker['salary']
        gender = worker['sex']

        # Apply the conditional statements
        if 7500 < salary < 30000 and gender == 'Female':
            worker['level'] = 'A5-F'
        elif 10000 < salary < 20000:
            worker['level'] = 'A1'
        else:
            worker['level'] = 'Other'  # Default level if no conditions are met

        # Print payment slips
        print(f"Pay slip: ID={worker['id']}, Name={worker['name']}, "
              f"Salary=${worker['salary']}, Sex={worker['sex']}, Level={worker['level']}")
    
    except Exception as e:
        # Exception handling for potential errors
        print(f"Error processing worker {worker['id']}: {e}")
